docs/tutorials/constraints.rst

```diff
@@ -0,0 +1,238 @@
+
+    from astroplan import Observer, FixedTarget
+    from astropy.time import Time
+    subaru = Observer.at_site("Subaru")
+    time_range = Time(["2015-08-01 06:00", "2015-08-01 12:00"])
+
+    # Read in the table of targets
+    from astropy.io import ascii
+    target_table = ascii.read('targets.txt')
+
+    # Create astroplan.FixedTarget objects for each one in the table
+    from astropy.coordinates import SkyCoord
+    import astropy.units as u
+    targets = [FixedTarget(coord=SkyCoord(ra=ra*u.deg, dec=dec*u.deg), name=name)
+               for name, ra, dec in target_table]
+
+In the above example, you may have noticed that constraints are assembled by
+making a list of calls to the initializers for classes like `AltitudeConstraint`
+and `AirmassConstraint`. Each of those constraint classes is subclassed from
+the abstract `Constraint` class, and the custom constraint that we're going to
+write must be as well.
+
+In this example, let's design our constraint to ensure that all targets must be
+within some number of degrees from Vega – we'll call it
+`VegaSeparationConstraint`. Two methods, `__init__` and `compute_constraint`
+must be written for our constraint to work:
+
+* The `__init__` method will accept the minimum and maximum acceptable separations
+  a target could have from Vega.
+
+* We'll also define a method `compute_constraints` which takes three arguments:
+  an array of times to test, an `Observer` object, and one or a list of
+  `FixedTarget` objects. `compute_constraints` will return a matrix of
+  booleans that describe whether or not each target meets the constraints.
+  The super class `Constraint` has a `__call__` method which will run your
+  custom class's `compute_constraints` method when you check if a target is
+  observable using `is_observable` or `is_always_observable`.
+
+Here's our `VegaSeparationConstraint` implementation::
+
+    from astroplan import Constraint, is_observable
+    from astropy.coordinates import Angle
+
+    class VegaSeparationConstraint(Constraint):
+        """
+        Constraint the separation from Vega
+        """
+        def __init__(self, min=None, max=None):
+            """
+            min : `~astropy.units.Quantity` or `None` (optional)
+                Minimum acceptable separation between Vega and target. `None`
+                indicates no limit.
+            max : `~astropy.units.Quantity` or `None` (optional)
+                Minimum acceptable separation between Vega and target. `None`
+                indicates no limit.
+            """
+            self.min = min
+            self.max = max
+
+        def compute_constraint(self, times, observer, targets):
+
+            # Vega's position is essentially unchanging, but if it were to be a
+            # moving target, we would need an array of coordinates for Vega as a
+            # function of time. Here we'll simulate that behavior with multiple
+            # copies of the Vega coordinate
```

Comment:
It will not work via broadcasting, which is why I did it. I discussed this example with @eteq and @adrn , but after coding this line, I started to doubt whether or not it's the best one to use. We wanted an example that was trivial to code and easy to understand, and I think this line doesn't meet the second criteria exactly.